tests/subsys/logging/log_stack/CMakeLists.txt

```
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(log_stack)
+
+target_sources(app PRIVATE src/main.c src/maincxx.cxx)
```

I believe if you update `test.yaml` to have tests where `CONFIG_CPLUSPLUS=y` then you can use this knowledge here, like:
```suggestion
target_sources_ifdef(CONFIG_CPLUSPLUS app PRIVATE src/maincxx.cxx)
target_sources_ifndef(CONFIG_CPLUSPLUS app PRIVATE src/main.c)
```

and is more aligned with how users should enable C++ support, [CONFIG_CPLUSPLUS](https://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_CPLUSPLUS.html)

and then you may cleanup as proposed here: https://github.com/zephyrproject-rtos/zephyr/pull/42686/commits/f9ac5ea65d8c9dccca4a775f49acf544f0bc21b3#r806674828

---

Done.